Automattic/jetpack
# VideoPress: Integrate connection error React component

#### Changes proposed in this Pull Request:
Integrate the `ConnectionErrorNotice` React component into VideoPress. This is part of a bigger project (1202697549597858-as) to provide consistent failed connection error reporting and an interface to reconnect (1202697549597858-as-1202697549597864) to all plugins and projects in the monorepo.

The changes will affect the following:
* VideoPress

#### Other information:


#### Jetpack product discussion
* p9dueE-5w8-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
1. Run `jetpack build packages/videopress`.
1. Enable the "Jetpack Debug Tools" plugin.
1. Go to Jetpack Debug > Jetpack Debug.
1. Enable "Broken token Utilities" and click "Save".
1. Go to Jetpack > VideoPress.
1. Confirm that no errors display below the "Jetpack VideoPress" heading.
1. Go to Jetpack Debug > Broken Token.
1. Click the "Set invalid blog token" button and the "Set invalid user tokens" button.
1. Run the following query on the database to clear connection error-related transients (if running the docker Jetpack environment, run `jetpack docker db`):
```delete from wp_options where option_name like '%jetpack_connection_error_reporting_gate_%';```
1. Go to Jetpack > My Jetpack.
1. Go to Jetpack > VideoPress.
1. You should see an error message as shown in the below screenshot:
![jetpack-connection-error-example-generic](https://user-images.githubusercontent.com/56307/193151665-e2cee674-2084-4d5c-9054-6d1313bb447a.png)
1. Click on "Restore Connection".
1. The "Reconnecting Jetpack" message appears with the animating wait image.
1. The "Reconnecting Jetpack" connection disappears.
1. You are taken to the Jetpack connection page.
